### PR TITLE
* Allow package downgrade if Pin-Priority exceeds 1000

### PR DESCRIPTION
**Pull request by h2x0r**
This PR allows unattended-upgrade to downgrade a package if a Pin-Priority exceeds 1000 and downgrade is allowed (Unattended-Upgrade::Allow-downgrade "true";).  It can be handy on many circumstances including:
1. Unattended rollback:
  - You have your own internal/private repository and thousands of hosts tracking changes on it. In case you want to roll back to previous version it is enough if you just remove latest version from repository
2. Software testing:
  - You can quickly roll back and forward only by adding or removing packages from repository

### CODE REVIEW

**Comment by rbalint** [comment]
@h2x0r This is a very interesting feature and aligns with the goal of fully supporting pinning. Please add test cases to let us keep the feature working easily.

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,9 @@
+unattended-upgrades (1.14+nmu1) unstable; urgency=low
+
+  * Allow package downgrade if Pin-Priority exceeds 1000
+
+ -- Kaupo Arulo <kaupoa@microsoft.com>  Tue, 08 Oct 2019 13:00:22 +0300
+
```
Comment (rbalint) [inline debian/changelog]: Please drop the change to the changelog, also please don't NMU it.

File: unattended-upgrade [hunk 2]
```diff
@@ -220,8 +220,9 @@ class UnattendedUpgradesCache(apt.Cache):
             self.adjust_candidate(self[pkg_name])
 
         if function == apt.package.Package.mark_upgrade \
-            and not pkg.is_upgradable:
-            if not apt_pkg.config.find_b("Unattended-Upgrade::Allow-downgrade", False):
+                and not pkg.is_upgradable:
```
Comment (rbalint) [inline unattended-upgrade]: Please squash this commit to the previous one to keep tests happy in every commit.

File: debian/changelog (hunk 1 above)
Comment (h2x0r) [reply debian/changelog]: Done

File: unattended-upgrade (hunk 2 above)
Comment (h2x0r) [reply unattended-upgrade]: Done. Sorry for a long delay...

**Pull request merged**